outputs.tf

```diff
@@ -223,3 +223,8 @@ output "username" {
   sensitive   = true
   value       = azurerm_kubernetes_cluster.main.kube_config[0].username
 }
+
+output "web_app_routing_identity" {
+  description = "The `azurerm_kubernetes_cluster`'s `web_app_routing_identity` block."
+  value       = try(azurerm_kubernetes_cluster.main.web_app_routing[0].web_app_routing_identity, null)
```

Comment:
According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L495-L514), `web_app_routing_identity` is a list of object, so I would recommend `[]` as default value instead of `null`, so the consumer can use `for` expression to iterate it:

```hcl
value       = try(azurerm_kubernetes_cluster.main.web_app_routing[0].web_app_routing_identity, [])
```

Otherwise, we'll receive an error when we're using `for` expression and this output is `null`:

```text
Error: Iteration over null value
```

Reply:
Done